src/Clerk.ts

```diff
@@ -156,17 +156,19 @@ export default class Clerk {
 
         Logger.debug(`sessionId from query: ${sessionId}`);
 
+        let session: (Session | null) = null;
+
         if (!sessionId) {
-          client = await this.clients.verifyClient(sessionToken);
-          sessionId = client.lastActiveSessionId;
-          Logger.debug(`lastActiveSessionId from client: ${sessionId}`);
+          Logger.debug('retrieving last active session for client');
+          session = await this.sessions.verify(sessionToken);
+        } else {
+          Logger.debug(`retrieving session ${sessionId}`);
+          session = await this.sessions.verifySession(
```

Comment:
🔧 It feels a bit confusing that we have `verify` and `verifySession` methods. May I suggest to use just one method and determine its behavior based on the parameters? For example we can do:

```this.sessions.verify({ sessionId?, sessionToken })```. 

If sessionId is applied it invokes the old endpoint, otherwise it invokes `sessions/verify` directly.

Reply:
@SokratisVidros Agreed, it feels weird. I personally would avoid adding this conditional logic into the `SessionsApi`...AFAICT (@yourtallness , correct me if I'm wrong), these APIs are just thin layers to abstract the actual API calls.
But I don't feel strongly about it, so if you guys think it's a good idea, happy to go for it.
Otherwise, how about renaming `sessions.verify` to something like `sessions.verifyClientSession` or `sessions.verifyClientToken`?

Reply:
I'd vote to add the if in the method.

Reply:
I proposed the conditional logic as it was too hard to find a good name. I guess we can co with the renaming for now. @colinclerk WDYT?

Reply:
Wouldn't it make sense to change the server api and have only a `/sessions/verify` endpoint with optional `session_id` param?

We can deprecate `/sessions/:id/verify`, keep it around for a couple of weeks, then remove it.

I'd prefer we don't add conditional logic to the api classes because the function names follow the openapi operation names.

Reply:
I don't think the openapi operation names should be a heavily weighted consideration here.  They're not exposed to the developer at present, and we should be thinking in terms of their experience.

`sessions.verify` reads to me like a more concise version of `sessions.verifySession`.  It's not intuitive to me that they would take different arguments.

Reply:
Maybe we can make /client/verify return the whole session object instead of just session IDs?  I think that would prevent the secondary call to `/sessions/:id/verify`.

Reply:
> Maybe we can make /client/verify return the whole session object instead of just session IDs?

@colinclerk Unfortunately, this is not a backwards compatible change, so will break users with previous versions of the SDKs.

> Wouldn't it make sense to change the server api and have only a /sessions/verify endpoint with optional session_id param?

@yourtallness Yep, we could definitely do that. 
@colinclerk @SokratisVidros What do you guys think? This basically means that we won't have any conditional logic in the SDKs, which is good IMO.

Reply:
I don't want to break rest by putting a `session_id` param on the /sessions endpoint.

Maybe an `expandSessions` param on /client/verify?

Reply:
@colinclerk Why are we breaking rest with `/sessions/:id/verify` Isn't that restful?

Reply:
After offline discussions, we decided to go for another approach.
The decision was to include the client's active sessions in the client response of the existing `/clients/verify` endpoint.
This way, the SDK doesn't have to do an additional call to retrieve the client's last touched session...instead it can retrieve it from the list of the client's active sessions.